aws/request/request_test.go

```diff
@@ -716,8 +717,11 @@ func TestEnforceShouldRetryCheck(t *testing.T) {
 
 	client := &http.Client{Transport: tp}
 
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		time.Sleep(5 * time.Millisecond)
+		wg.Wait()
```

Comment:
I suggest just a empty `select{}` for this instead of creating a `WaitGroup`.  Adding a comment line stating why this should wait for ever would also help clear up any confusion.